### PR TITLE
Add size to base adapter

### PR DESCRIPTION
**Pull request by ejaypcanaria**

### CODE REVIEW

**Comment by bradleypriest** [comment]
Tests?

**Comment by bradleypriest** [comment]
Changelog?

**Comment by bradleypriest** [comment]
Also I was thinking more along the lines of 

```ruby
def size
  @pagination['total_records'] || count
end
```

That way they can make another API call and then call size again and get the new number

**Comment by ejaypcanaria** [comment]
I tried that but it's failing because `@pagination` is nil on first run.

**Comment by bradleypriest** [comment]
@defined?(:@pagination) instead maybe?

**Comment by ejaypcanaria** [comment]
@bradleypriest is this good to merge?

File: lib/gecko/record/base_adapter.rb [hunk 1]
```diff
@@ -128,6 +128,18 @@ def count
         @pagination['total_records']
       end
 
+      # Return a memoized total count for a record type
```
Comment (bradleypriest) [inline lib/gecko/record/base_adapter.rb]: This isn't correct anymore is it?

Comment (ejaypcanaria) [inline lib/gecko/record/base_adapter.rb]: I believe it's still correct since it's not going to call `count` if `@pagination['total_records']` has a value.

Comment (bradleypriest) [inline lib/gecko/record/base_adapter.rb]: Sadly the Rails equivalent isn't so descriptive in it's naming
https://github.com/rails/rails/blob/master/activerecord/lib/active_record/relation.rb#L256

Comment (ejaypcanaria) [inline lib/gecko/record/base_adapter.rb]: Oh, that's sad :(

**Comment by ejaypcanaria** [comment]
@bradleypriest good to merge?

**Comment by bradleypriest** [comment]
I still believe the method description is wrong. Should I just update it?

**Comment by ejaypcanaria** [comment]
No worries, I'll fix the description.

**Comment by ejaypcanaria** [comment]
@bradleypriest let me know once the description is good. If still not good, I'll be happy for you to update it :grin:

**Comment by bradleypriest** [comment]
Updated the description, can you squash it down into 1 commit and we'll be good to go

**Pull request merged**